### PR TITLE
Added rest proxy operator

### PR DESCRIPTION
**Pull request by vibhujain**
Signed-off-by: Vibhu Jain <vibhujain.j@gmail.com>

### CODE REVIEW

File: repository/rest-proxy/operator/params.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: kudo.dev/v1beta1
+parameters:
+  - name: HOST_NAME
+    description: "Host name used to generate absolute URLs in responses"
+    default: "localhost"
+  - name: SCHEMA_REGISTRY_URL
+    description: "The base URL for Confluent Schema Registry that should be used by the Avro serializer"
+    default: "http://localhost:8081/"
```
Comment (zmalik) [inline repository/rest-proxy/operator/params.yaml]: let's use what would be default schema registry URL of the schema registry operator

File: repository/rest-proxy/operator/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+  labels:
+    app: rest-proxy
+spec:
+  replicas: 1
```
Comment (zmalik) [inline repository/rest-proxy/operator/templates/deployment.yaml]: let's parametrize that

File: repository/rest-proxy/operator/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          name: rest-proxy
+        env:
+        - name: KAFKA_REST_SCHEMA_REGISTRY_URL 
+          value: {{ .Params.SCHEMA_REGISTRY_URL }}
+        - name: KAFKA_REST_HOST_NAME
+          value: {{ .Params.HOST_NAME }}
+        - name: KAFKA_REST_BOOTSTRAP_SERVERS
+          value: {{ .Params.KAFKA_BOOTSTRAP_SERVERS }}
+        livenessProbe:
+          initialDelaySeconds: 30
```
Comment (zmalik) [inline repository/rest-proxy/operator/templates/deployment.yaml]: let's parametrize these options, with the default being these ones. As bootstrap can take time depending on the state of Kafka

File: repository/rest-proxy/operator/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: rest-proxy
+    spec:
+      containers:
+      - name: rest-proxy
+        image: mesosphere/cp-kafka-rest:5.3.0
```
Comment (zmalik) [inline repository/rest-proxy/operator/templates/deployment.yaml]: as we talked, lets test with confluent image and not mesosphere image

**Review by zmalik — CHANGES_REQUESTED** [review]
looking good! 
leaving few comments that should be easy to address

File: repository/rest-proxy/operator/params.yaml (hunk 1 above)
Comment (zmalik) [reply repository/rest-proxy/operator/params.yaml]: with this I mean, if a user installs zookeeper/kafka/schema-registry/rest-proxy in the same namespace, it should work with the default parameters

Comment (vibhujain) [reply repository/rest-proxy/operator/params.yaml]: Done. Changed as suggested.

File: repository/rest-proxy/operator/templates/deployment.yaml (hunk 3 above)
Comment (vibhujain) [reply repository/rest-proxy/operator/templates/deployment.yaml]: Done. Added parameters to configure this.

File: repository/rest-proxy/operator/templates/deployment.yaml (hunk 4 above)
Comment (vibhujain) [reply repository/rest-proxy/operator/templates/deployment.yaml]: Done. Changed the image to Confluentic one, and now using the latest version v5.3.2.

File: repository/rest-proxy/operator/templates/deployment.yaml (hunk 2 above)
Comment (vibhujain) [reply repository/rest-proxy/operator/templates/deployment.yaml]: Done. Added parameters to configure this.

File: repository/rest-proxy/operator/operator.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+apiVersion: kudo.dev/v1beta1
+name: "rest-proxy"
+version: "0.1.0"
+kudoVersion: 0.8.0
+kubernetesVersion: 1.15.0
+appVersion: 5.3.0
```
Comment (zmalik) [inline repository/rest-proxy/operator/operator.yaml]: let's update the app version to the one used in image

**Review by zmalik — CHANGES_REQUESTED** [review]

File: repository/rest-proxy/operator/operator.yaml (hunk 5 above)
Comment (vibhujain) [reply repository/rest-proxy/operator/operator.yaml]: Done. Thanks for the correction!

**Review by zmalik — APPROVED** [review]
LGTM
thanks @vibhujain

**Comment by guenter** [comment]
This operator should be called "kafka-rest-proxy" to make it clear that it is for Kafka only, not a generic REST proxy. That's also the official name for it (see https://github.com/confluentinc/kafka-rest).

**Comment by guenter** [comment]
Confluent also calls it "Confluent REST Proxy" so maybe "confluent-rest-proxy" is a better name to make it consistent with Schema Registry (see #170)

**Comment by zmalik** [comment]
both packages renamed to `confluent-rest-proxy` and `confluent-schema-registry` and updated in KUDO repository also

**Pull request merged**